sdk/python/kfp/compiler/_container_builder.py

```diff
@@ -47,6 +53,26 @@ def __init__(self, gcs_staging=None, namespace=None):
     from ._gcs_helper import GCSHelper
     GCSHelper.create_gcs_bucket_if_not_exist(gcs_bucket)
 
+    # Configure the GCR image tag
+    # Kubeflow jupyter notebook is injected with a variable as an ID
+    if gcr_image_tag is None:
+      KF_NOTEBOOK_ENV = 'NB_PREFIX'
+      nb_id = ''
+      try:
+        if KF_NOTEBOOK_ENV in os.environ.keys():
+          nb_id = os.environ[KF_NOTEBOOK_ENV]
+        else:
+          nb_id = self._get_instance_id()
+      except:
+        raise ValueError('Please provide the gcr_image_tag.')
```

Comment:
... when running outside of the KFP cluster.

Reply:
It currently works for ai platform notebook and kf notebook where the project id is available. Otherwise, users need to provide one.